### PR TITLE
Improve timing tests

### PR DESCRIPTION
**Pull request by kichristensen**
Timer tests has be very unstable, this is improved a lot in this PR

### CODE REVIEW

File: src/Okanshi/Timers.fs [hunk 1]
```diff
@@ -4,20 +4,50 @@ open System
...
+    abstract Start : unit -> unit
+    abstract Stop : unit -> unit
+    abstract Time : Func<'T> -> ('T * int64)
+    abstract Time : Action -> int64
+
+type SystemStopwatch() =
+    let stopwatch = new Stopwatch()
+
+    let time' f =
+        stopwatch.Start()
```
Comment (ChristianTrolleMikkelsen) [inline src/Okanshi/Timers.fs]: Reentrant? Stopwatch do have reset/restart methods. Are you sure this is correct?

File: tests/Okanshi.Tests/BasicTimerTest.cs [hunk 2]
```diff
@@ -160,38 +164,41 @@ public void Manual_timing_sets_max()
         {
             timer.GetCount();
             var okanshiTimer = timer.Start();
-            Thread.Sleep(50);
+            stopwatch.IsRunning.Returns(true);
+            stopwatch.ElapsedMilliseconds.Returns(50);
```
Comment (ChristianTrolleMikkelsen) [inline tests/Okanshi.Tests/BasicTimerTest.cs]: Missing stopwatch.Time(Arg.Any<Action>()).Returns(50); ???

File: tests/Okanshi.Tests/BasicTimerTest.cs [hunk 3]
```diff
@@ -160,38 +164,41 @@ public void Manual_timing_sets_max()
...
         }
 
         [Fact]
         public void Manual_timing_sets_min()
         {
             timer.GetCount();
             var okanshiTimer = timer.Start();
-            Thread.Sleep(50);
+            stopwatch.IsRunning.Returns(true);
+            stopwatch.ElapsedMilliseconds.Returns(50);
```
Comment (ChristianTrolleMikkelsen) [inline tests/Okanshi.Tests/BasicTimerTest.cs]: Missing stopwatch.Time(Arg.Any<Action>()).Returns(50); ???

File: tests/Okanshi.Tests/BasicTimerTest.cs [hunk 4]
```diff
@@ -160,38 +164,41 @@ public void Manual_timing_sets_max()
...
         }
 
         [Fact]
         public void Manual_timing_sets_total_time()
         {
             timer.GetTotalTime();
             var okanshiTimer = timer.Start();
-            Thread.Sleep(500);
+            stopwatch.IsRunning.Returns(true);
+            stopwatch.ElapsedMilliseconds.Returns(50);
```
Comment (ChristianTrolleMikkelsen) [inline tests/Okanshi.Tests/BasicTimerTest.cs]: Missing stopwatch.Time(Arg.Any<Action>()).Returns(50); ???

File: tests/Okanshi.Tests/BasicTimerTest.cs (hunk 2 above)
Comment (kichristensen) [reply tests/Okanshi.Tests/BasicTimerTest.cs]: No as this is a manual timing it is not needed

File: tests/Okanshi.Tests/BasicTimerTest.cs (hunk 3 above)
Comment (kichristensen) [reply tests/Okanshi.Tests/BasicTimerTest.cs]: No as this is a manual timing it is not needed

File: tests/Okanshi.Tests/BasicTimerTest.cs (hunk 4 above)
Comment (kichristensen) [reply tests/Okanshi.Tests/BasicTimerTest.cs]: No as this is a manual timing it is not needed

File: src/Okanshi/Timers.fs (hunk 1 above)
Comment (kichristensen) [reply src/Okanshi/Timers.fs]: I've made the stopwatch ctor to internal as it is not meant to be create outside Okanshi, and therefore it won't be used reentrantly

**Comment by kichristensen** [comment]
@kbilsted The timer is removed because of very few use cases and the maintenance required to maintain it. The best migration path is to create it yourself

**Pull request merged**